supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigAsyncJobExecutionActor.scala

```diff
@@ -246,4 +250,50 @@ class DispatchedConfigAsyncJobExecutionActor(override val standardParams: Standa
   override def killArgs(job: StandardAsyncJob): SharedFileSystemCommand = {
     jobScriptArgs(job, "kill", KillTask)
   }
+
+  protected lazy val exitCodeTimeout: Option[Int] =
+    configurationDescriptor.backendConfig.as[Option[Int]](ExitCodeTimeoutConfig).map(math.abs)
+
+  override def pollStatus(handle: StandardAsyncPendingExecutionHandle): SharedFileSystemRunStatus = {
+    handle.previousStatus match {
+      case None =>
+        // Is not set yet the status will be set default to running
+        SharedFileSystemRunStatus("Running")
+      case Some(s) if (s.status == "Running" || s.status == "WaitingForReturnCode") && jobPaths.returnCode.exists =>
+        // If exitcode file does exists status will be set to Done always
+        SharedFileSystemRunStatus("Done")
+      case Some(s) if s.status == "Running" =>
+        // Exitcode file does not exist at this point, checking is jobs is still alive
+        if (exitCodeTimeout.isEmpty) s
+        else if (isAlive(handle.pendingJob).getOrElse(true)) s
```

Comment:
Are you happy that this `getOrElse(true)` is safe?

Reply:
no not really but I don't want to raise an exception because sometimes this kind of calls just fail. SGE is not so stable ;)

Reply:
ye that should be fine, does not happen that often.